### DISCUSSION TITLE
Welcome to CS50 Discussions!

### DISCUSSION THREAD

**Discussion by mareksuscak** [mareksuscak/cs50#11] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mattchia** [comment]
Hello! I'm not sure if this is the right place to ask this but here goes.

I am currently working on CS50's pset2/caesar and I was wondering why I have to check argc before argv. If I check argv first, I'll receive a segmentation fault.


This is the code that works!
![image](https://user-images.githubusercontent.com/75569136/102459397-810c7180-4080-11eb-9236-cc4354da1f8d.png)


This is the code that doesn't....
![image](https://user-images.githubusercontent.com/75569136/102459480-9bdee600-4080-11eb-8d2a-a0d861d1ea66.png)

Thanks in advance!

**Comment by mareksuscak** [comment]
@mattchia First and foremost, welcome to the CS50x community! To your question, if you don't change the sequencing of the aforementioned operations, you could easily run into the segmentation fault issue which is typically caused by accessing elements in an array that are outside of bounds of the array which often happens when you do not provide the required number of arguments on the command line. [Here's a list of other common reasons for Segmentation Faults in C and C++](https://www.tutorialspoint.com/List-of-Common-Reasons-for-Segmentation-Faults-in-C-Cplusplus).
